src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/attachscanneddocs/AttachScannedDocsAboutToSubmitHandlerTest.java

```diff
@@ -67,4 +69,61 @@ public void givenAnAttachScannedDocsEventAndAvEvidenceAlreadyOnCase_thenSetEvide
+                        .url(DocumentLink.builder().documentUrl("original url")
+                                .documentBinaryUrl("original binary url").documentFilename("original scanned doc").documentHash("hash original").build())
+                        .type("Other")
+                        .scannedDate("20 Jun 2023")
+                        .build()
+        ).build();
+
+        ScannedDocument document2 = ScannedDocument.builder().value(
+                ScannedDocumentDetails.builder()
+                        .url(DocumentLink.builder().documentUrl("original url")
+                                .documentBinaryUrl("original binary url").documentFilename("original scanned doc").documentHash("hash original").build())
+                        .type("Other")
+                        .scannedDate("20 Jun 2023")
+                        .build()
+        ).build();
+
+        ScannedDocument document3 = ScannedDocument.builder().value(
+                ScannedDocumentDetails.builder()
+                        .editedUrl(DocumentLink.builder().documentUrl("edited url2")
+                                .documentBinaryUrl("edited binary url2").documentFilename("edited scanned doc2").documentHash("hash edited2").build())
+                        .url(DocumentLink.builder().documentUrl("original url2")
+                                .documentBinaryUrl("original binary url2").documentFilename("original scanned doc2").documentHash("hash original2").build())
+                        .type("Other")
+                        .scannedDate("24 Jun 2023")
+                        .build()
+        ).build();
+
+        ScannedDocument document4 = ScannedDocument.builder().value(
+                ScannedDocumentDetails.builder()
+                        .url(DocumentLink.builder().documentUrl("original url2")
+                                .documentBinaryUrl("original binary url2").documentFilename("original scanned doc2").documentHash("hash original2").build())
+                        .type("Other")
+                        .scannedDate("24 Jun 2023")
+                        .build()
+        ).build();
+
+        CaseData caseData = SscsCaseData.builder().scannedDocuments(List.of(document2, document4)).build();
+        CaseData caseDataBefore = SscsCaseData.builder().scannedDocuments(List.of(document1,document3)).build();
+
+        CaseDetails<? extends CaseData> caseDetails = new CaseDetails<>(1L, "", null, caseData, null, null);
+        CaseDetails<? extends CaseData> caseDetailsBefore = new CaseDetails<>(1L, "", null, caseDataBefore, null, null);
+
+        Callback<SscsCaseData> testCallback = new Callback(caseDetails, Optional.of(caseDetailsBefore), EventType.ATTACH_SCANNED_DOCS, false);
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, testCallback, USER_AUTHORISATION);
```

Comment:
in reality, the response shouldn't have the documents from the caseDataBefore, because the scannedDocuments event might add some extra documents.
what we need to assert in the response is that it still has the documents that were there before the event was called, and that the edited URLs weren't lost